forge-game/src/main/java/forge/game/ability/effects/DamageEachEffect.java

```diff
@@ -58,10 +55,20 @@ protected String getStackDescription(SpellAbility sa) {
     public void resolve(SpellAbility sa) {
         final Card card = sa.getHostCard();
         final Game game = card.getGame();
+        final String num = sa.getParamOrDefault("NumDmg", "X");
 
         FCollectionView<Card> sources = game.getCardsIn(ZoneType.Battlefield);
         if (sa.hasParam("ValidCards")) {
             sources = CardLists.getValidCards(sources, sa.getParam("ValidCards"), sa.getActivatingPlayer(), card, sa);
+        } else if (sa.hasParam("DefinedDamagers")) {
+            List<Card> defined = AbilityUtils.getDefinedCards(card, sa.getParam("DefinedDamagers"), sa);
```

Comment:
wouldn't that just be sources = Defined? or just checking if Defined is inZoneBattlefield?

there is no need to get all cards in Battlefield, and then overwrite them from DefinedDamagers anyway